test foo java io exception ioexception java io exception ioexception